lib/recipe/widgets/ingredient_form.dart

```diff
@@ -25,259 +42,354 @@ class IngredientForm extends StatelessWidget {
-                            .ingredients[ingredientIndex].name.invalid
-                        ? 'Enter an ingredient name e.g. Tomato'
-                        : null,
-                  ),
-                  onChanged: (value) {
-                    context.read<RecipeBloc>().add(IngredientNameChanged(
-                        ingredient: state
-                            .recipes[recipeIndex].ingredients[ingredientIndex],
-                        ingredientName: value,
-                        recipe: state.recipes[recipeIndex]));
-                  },
-                  textInputAction: TextInputAction.next,
+        padding: const EdgeInsets.all(8.0),
+        child: Column(
+          children: [
+            FutureBuilder(
+                future: _ingredients,
+                builder:
+                    (BuildContext context, AsyncSnapshot<String> snapshot) {
+                  if (snapshot.hasData) {
+                    return DropdownSearch<String>(
+                        dropdownSearchDecoration: const InputDecoration(
+                          icon: Icon(Icons.food_bank_rounded),
+                          labelText: 'Ingredient name',
+                          helperText: 'Ingredient name e.g. Potato',
+                        ),
+                        mode: Mode.MENU,
+                        showSelectedItems: true,
+                        showSearchBox: true,
+                        items: json.decode(snapshot.data!).keys.toList(),
+                        onChanged: (String? answer) => context
+                            .read<RecipeBloc>()
+                            .add(IngredientNameChanged(
+                                ingredient: state.recipes[widget.recipeIndex]
+                                    .ingredients[widget.ingredientIndex],
+                                ingredientName: answer!,
+                                recipe: state.recipes[widget.recipeIndex])),
+                        selectedItem: state.recipes[widget.recipeIndex]
+                            .ingredients[widget.ingredientIndex].name);
+                  } else {
+                    return DropdownSearch<String>(
+                        dropdownSearchDecoration: const InputDecoration(
+                          icon: Icon(Icons.food_bank_rounded),
+                          labelText: 'Ingredient name',
+                          helperText: 'Ingredient name e.g. Potato',
+                        ),
+                        enabled: false,
+                        mode: Mode.MENU,
+                        showSelectedItems: true,
+                        selectedItem: state.recipes[widget.recipeIndex]
+                            .ingredients[widget.ingredientIndex].name);
+                  }
+                }),
+            Visibility(
+              visible: (state.recipes[widget.recipeIndex]
+                      .ingredients[widget.ingredientIndex].name ==
+                  "Other (please specify)"),
```

Comment:
This is again one place where it is kind of diffucult to understand without reading other bits of code what the otherName is – maybe this logic could be refactored to be more clear that the user has not selected any default ingredient but has instead made a custom ingredient. Maybe:
```dart
visible: (state.recipes[widget.recipeIndex]
                      .ingredients[widget.ingredientIndex]
                      .customNameChosen),
```